public/components/map_container/map_container.tsx

```diff
@@ -168,17 +169,27 @@ export const MapContainer = ({
     if (isUpdatingLayerRender || isReadOnlyMode) {
       if (selectedLayerConfig) {
         if (baseLayerTypeLookup[selectedLayerConfig.type]) {
-          handleBaseLayerRender(selectedLayerConfig, maplibreRef, undefined);
+          handleBaseLayerRender(selectedLayerConfig, maplibreRef);
         } else {
           updateIndexPatterns();
-          handleDataLayerRender(selectedLayerConfig, mapState, services, maplibreRef, undefined);
+          handleDataLayerRender(selectedLayerConfig, mapState, services, maplibreRef);
         }
       } else {
         renderDataLayers(layers, mapState, services, maplibreRef, timeRange, filters, query);
         renderBaseLayers(layers, maplibreRef);
+        // Because of async layer rendering, layers order is not guaranteed, so we need to order layers
+        // after all layers are rendered.
+        maplibreRef.current!.once('idle', () => {
+          orderLayers(layers, maplibreRef.current!);
```

Comment:
Can you refactor anonymous method to named method?

Reply:
Updated